docs/adr/ADR-003_Acceptable_use_of_GitHub_PAT_and_Apps_for_authN_and_authZ.md

```diff
@@ -0,0 +1,225 @@
+
+- **Server-to-server communication**: Unlike users, GitHub Apps have their own identities and can perform actions directly on a repository without a user action triggering them. They are associated with the GitHub account (individual or organisation) that owns the app, not necessarily the account that installed the app. In this model the GitHub App can perform actions based on the permissions it was given during setup. These permissions are separate from any user permissions and allow the app to interact with the GitHub API directly. For example, an app might be set up to automatically run a test suite whenever code is pushed to a repository. This action would happen regardless of which user pushed the code.
+
+This method of authentication and authorisation is <u>intended for the engineering teams</u> to implement and support automated processes. Setting up the [GitHub OAuth App](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/authorizing-oauth-apps) access is outside the scope of this document as this mechanism should not be employed in the context of development process automation.
+
+### Rationale
+
+This guide describes the essence of the fundamental aspects of GitHub authentication and authorisation mechanisms along with the common use cases identified by the GitHub organisation administrators of the NHS England.
+
+## Notes
+
+### GitHub App setup
+
+To be executed by a GitHub organisation administrator:
+
+- Identify the GitHub repository name for which the team has requested a GitHub App integration
+- Create a shared email address [england.[repository-name]-app@nhs.net](england.[repository-name]-app@nhs.net) by filling in the `New shared mailbox request` form using the Internal Portal (ServiceNow)
+  - Delegate access to this mailbox for the GitHub organisation owners, administrators and the engineering team
+- Create a GitHub bot account named `[repository-name]-app` using the email address mentioned above
+  - Use the `nhs.net` email address as the default and only email
+  - Set the email address as private
+  - Make profile private and hide any activity
+  - Block command line pushes that expose email
+  - Set up commit signing
+  - Flag unsigned commits as unverified
+- [Register new GitHub App](https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/registering-a-github-app) under the `[repository-name]-app` bot account with a name following a pattern that could include information like `[Team] [Repository Name] [Purpose]`, which would help to search for and identify owners of the app once it is installed within the GitHub organisation
+  - Make note of the `App ID`
+  - Generate and store securely a `Private key` for the app
+  - Provide a `Homepage URL` to the repository this app will operate on
+  - Make this app `public`
+  - Set the relevant `Repository permissions` based on the team's requirements. There should be no organisation or account permissions set at all
```

Comment:
The bot account should not be added to the organisation, therefore no seat is going to be taken. It is used as an identity, but the authentication and authorisation is done via the app. This will avoid giving the bot admin permission to  the repo but commits can still be signed. Access is controlled only via the app.

I'll expand on that in the ADR when I have a moment.

Reply:
I've added a clear note on this point. Thanks